WalletWasabi.Tests/UnitTests/AllFeeEstimateTests.cs

```diff
@@ -72,17 +76,160 @@ public void AllFeeEstimateHandlesInconsistentData()
 
 			estimations = new Dictionary<int, int>
 			{
-				{ 5, 1000 },
+				{ 18, 1000 },
 				{ 3, 21 },
 				{ 2, 20 },
 				{ 100, 100 },
-				{ 4, 4 },
+				{ 6, 4 },
 			};
 
 			allFee = new AllFeeEstimate(EstimateSmartFeeMode.Conservative, estimations, true);
 			Assert.Equal(2, allFee.Estimations.Count);
 			Assert.Equal(estimations[2], allFee.Estimations[2]);
-			Assert.Equal(estimations[4], allFee.Estimations[4]);
+			Assert.Equal(estimations[6], allFee.Estimations[6]);
+		}
+
+		[Fact]
+		public async Task RpcNotEnoughEstimationsAsync()
+		{
+			var rpc = new MockRpcClient();
+			rpc.OnGetBlockchainInfoAsync = async () =>
+				await Task.FromResult(new BlockchainInfo
+				{
+					Blocks = 100,
+					Headers = 100
+				});
+			rpc.OnGetPeersInfoAsync = async () =>
+				await Task.FromResult(Array.Empty<PeerInfo>());
+			rpc.OnGetMempoolInfoAsync = async () =>
+				await Task.FromResult(new MemPoolInfo
+				{
+					MemPoolMinFee = 0.00001000 // 1 s/b (default value)
+				});
+			rpc.OnEstimateSmartFeeAsync = (target, _) =>
+				throw new NoEstimationException(target);
+
+			await Assert.ThrowsAsync<NoEstimationException>(async () => await rpc.EstimateAllFeeAsync(EstimateSmartFeeMode.Conservative));
+		}
+
+		[Fact]
+		public async Task RpcFailuresAsync()
+		{
+			var rpc = new MockRpcClient();
+			rpc.OnGetBlockchainInfoAsync = () =>
+				throw new RPCException(RPCErrorCode.RPC_CLIENT_NOT_CONNECTED, "Error-GetBlockchainInfo", null);
+
+			rpc.OnEstimateSmartFeeAsync = (target, _) =>
+				throw new RPCException(RPCErrorCode.RPC_CLIENT_NOT_CONNECTED, "Error-EstimateSmartFee", null);
+
+			rpc.OnGetMempoolInfoAsync = async () =>
+				await Task.FromResult(new MemPoolInfo
+				{
+					MemPoolMinFee = 0.00001000 // 1 s/b (default value)
+				});
+
+			var ex = await Assert.ThrowsAsync<RPCException>(async () => await rpc.EstimateAllFeeAsync(EstimateSmartFeeMode.Conservative));
+			Assert.Equal(RPCErrorCode.RPC_CLIENT_NOT_CONNECTED, ex.RPCCode);
+			Assert.Equal("Error-EstimateSmartFee", ex.Message);
+		}
+
+		[Fact]
+		public async Task ToleratesRpcFailuresAsync()
+		{
+			var rpc = CreateAndConfigureRpcClient(
+				estimator: target => target switch
+				{
+					2 => new FeeRate(100m),
+					3 => throw new RPCException(RPCErrorCode.RPC_INTERNAL_ERROR, "Error", null),
+					5 => new FeeRate(89m),
+					6 => new FeeRate(75m),
+					8 => new FeeRate(70m),
+					_ => throw new NoEstimationException(target)
+				}
+			);
+
+			var allFee = await rpc.EstimateAllFeeAsync(EstimateSmartFeeMode.Conservative);
+			Assert.Equal(2, allFee.Estimations.Count);
+			Assert.False(allFee.Estimations.ContainsKey(3));
```

Comment:
In theory, this can potentially miss changes in output of `EstimateAllFeeAsync` as you assert what is NOT true.

Reply:
This is because I am verifying that in case the rpc call fails for target `3` (an expected target) it is not included. If for some reason the target `3` is added then we have a bug somewhere.